packages/labs/motion/src/view-transition-update.ts

```diff
@@ -0,0 +1,48 @@
+import {LitElement} from 'lit';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+/** Extend a LitElement to use view transitions (if available) on DOM updates */
```

Comment:
I think we need to extend these docs to explain the choices this makes, especially what timing this operates on (waits for ViewTransition.finished) and how this effects the LitElement lifecycle (it should delay update complete until the transition is finished).

Reply:
@justinfagnani hmm, am I missing a step? We need to tell the transition once the DOM update has finished before it runs the anim, and `performUpdate` isn't and shouldn't be `async`, but we'd potentially want `updateComplete` to only return once the anim has finished?

Is there a promise before `updateComplete` I can pass to `startViewTransition` instead?